### PR TITLE
Modal Refinements Update

### PR DESCRIPTION
**Pull request by RaymondDashWu**
Updated version of [this PR](https://github.com/uber/hubble.gl/pull/78). Previous PR was not merged and now touches parts of the codebase I didn't write.

# Changelog
- Preview button changed from button to icon below map preview
- Aspect ratio added to constants.js
- Text from "Resolution" simplified by dropping aspect ratios. Ex: "Good 16:9 (720p)" to "Good (720p)"
- "Resolution" dropdown changed to "Quality"
- Aspect Ratio moved to Export Settings
- Aspect ratio option added. Only resolutions of selected aspect ratio will appear. In addition, map preview will change based on aspect ratio selected
- Refactored many styled div to styled components and put into styled-components.js
- Added styling changes to modal to look more consistent with Kepler's current modals
- Fixed bug where rendering feedback overlapped with spinner.
- Tabs renamed from Edit/Export to Animation/Settings. Filenames changed as well as references

image (2)
image

### CODE REVIEW

**Review by chrisgervang — APPROVED** [review]

File: examples/kepler-integration/src/app.js [hunk 1]
```diff
@@ -107,8 +107,8 @@ const WindowSize = styled.div`
 `;
 
 // Custom localizations needed for export video modal
-messages.en['exportVideoModal.edit'] = 'Edit';
-messages.en['exportVideoModal.export'] = 'Export';
+messages.en['exportVideoModal.animation'] = 'Animation';
+messages.en['exportVideoModal.export'] = 'Settings';
```
Comment (macrigiuseppe) [inline examples/kepler-integration/src/app.js:111]: should you also change export key to be `'exportVideoModal.settings'`

File: modules/react/src/components/export-video/modal-tab-export.js [hunk 2]
```diff
@@ -31,10 +32,26 @@ function ExportTab({settings, resolution}) {
               searchable={false}
               onChange={settings.setMediaType}
             />
-            <StyledLabelCell>Resolution</StyledLabelCell>
+            <StyledLabelCell>Ratio</StyledLabelCell>
+            <ItemSelector
+              selectedItems={aspRatio}
+              options={['4:3', '16:9']}
```
Comment (macrigiuseppe) [inline modules/react/src/components/export-video/modal-tab-export.js:38]: can we have the list of options as a constant variable?

**Comment by macrigiuseppe** [comment]
@chrisgervang Few comments:
- if the default tab is settings, why isn't the first tab?
- The style for the active tab seems off. Check this out
Screen Shot 2021-08-06 at 7 44 03 PM
I think the bottom border should go underneath the current active tab

File: modules/react/src/components/export-video/constants.js [hunk 3]
```diff
@@ -51,39 +51,45 @@ export const FORMATS = [
 export const RESOLUTIONS = [
   {
     value: '960x540',
-    label: 'Good 16:9 (540p)',
+    label: 'Good (540p)',
     width: 960,
-    height: 540
+    height: 540,
+    aspectRatio: '16:9'
```
Comment (macrigiuseppe) [inline modules/react/src/components/export-video/constants.js:57]: i would create a list of ratios to use:
```
export const RATIOS = ['4:3', '16:9']
```
or
```
export const RATIOS = {'4_3': '4:3', '16_9': '16:9'];
```
and re-use it

File: modules/react/src/components/export-video/constants.js [hunk 4]
```diff
@@ -51,39 +51,45 @@ export const FORMATS = [
...
   },
   {
     value: '1280x720',
-    label: 'High 16:9 (720p)',
+    label: 'High (720p)',
     width: 1280,
-    height: 720
+    height: 720,
+    aspectRatio: '16:9'
```
Comment (macrigiuseppe) [inline modules/react/src/components/export-video/constants.js:64]: and you can use:
```
aspectRatio={RATIOS[1]}
```
or
```
aspectRatio={RATIOS['4_3']}
```

File: examples/kepler-integration/src/app.js (hunk 1 above)
Comment (RaymondDashWu) [reply examples/kepler-integration/src/app.js]: Good catch! I see that I forgot to make that consistent in several other places as well. Fixed w/ latest commit

**Comment by RaymondDashWu** [comment]
> @chrisgervang Few comments:
> 
> * if the default tab is settings, why isn't the first tab?
> * The style for the active tab seems off. Check this out
> 
> Screen Shot 2021-08-06 at 7 44 03 PM
> 
> I think the bottom border should go underneath the current active tab

Sorry for the confusion. That note was from the previous PR that didn't get merged in. Along the way we decided that we wanted the flow to allow users to select the type of animation before picking settings like resolution and aspect ratio. The currently active tab in that screenshot is Animation. I think that addresses your concern but let me know if I misinterpreted.

File: modules/react/src/components/export-video/modal-tab-export.js (hunk 2 above)
Comment (RaymondDashWu) [reply modules/react/src/components/export-video/modal-tab-export.js]: Fixed with latest commit. Good catch!

File: modules/react/src/components/export-video/constants.js (hunk 3 above)
Comment (RaymondDashWu) [reply modules/react/src/components/export-video/constants.js]: Fixed

File: modules/react/src/components/export-video/constants.js (hunk 4 above)
Comment (RaymondDashWu) [reply modules/react/src/components/export-video/constants.js]: Fixed

**Comment by RaymondDashWu** [comment]
Added some new bugs that I previously didn't catch. Will work on fixing them over the weekend.

**Pull request merged**